### PR TITLE
feat: enable setting a snapshot description for EBS surrogate

### PR DESCRIPTION
**Pull request by gnought**
Similar to EBS volume, this PR extends the support of setting Snapshot description in EBS surrogate

### CODE REVIEW

File: builder/ebssurrogate/step_snapshot_volumes.go [hunk 1]
```diff
@@ -68,13 +68,15 @@ func (s *StepSnapshotVolumes) snapshotVolume(ctx context.Context, deviceName str
 			ResourceType: aws.String("snapshot"),
 			Tags:         snapshotTags,
 		}
-
 		tagSpecs = append(tagSpecs, snapTags)
 	}
-
+	description := s.SnapshotDescription
+	if description == "" {
+		description = fmt.Sprintf("Packer: %s", time.Now().String())
```
Comment (lbajolet-hashicorp) [inline builder/ebssurrogate/step_snapshot_volumes.go:75]: I would suggest leaving the description empty be default, if only not to change the current behaviour by default, but also as the proposed description does not add information to the snapshot itself.

File: builder/ebsvolume/step_snapshot_ebs_volumes.go [hunk 2]
```diff
@@ -55,11 +56,14 @@ func (s *stepSnapshotEBSVolumes) Run(ctx context.Context, state multistep.StateB
 					ResourceType: aws.String("snapshot"),
 					Tags:         tags,
 				}
-
+				description := configVolumeMapping.SnapshotDescription
+				if description == "" {
+					description = fmt.Sprintf("Packer: %s", time.Now().String())
```
Comment (lbajolet-hashicorp) [inline builder/ebsvolume/step_snapshot_ebs_volumes.go:61]: Same here, I would suggest leaving the description empty by default.

**Review by lbajolet-hashicorp — COMMENTED** [review]
Hi @gnought,

Thanks for the PR!
I left a small comment on the default value, I think the empty default is good enough that we don't need to generate one if not provided.

Aside from that nit, LGTM! Pre-approving so it is not a blocker later.

**Review by lbajolet-hashicorp — APPROVED** [review]

**Comment by lbajolet-hashicorp** [comment]
Note: I see you haven't signed the CLAs, would you be able to do so? If not we won't be able to accept the contribution for legal reasons, sorry about that.

File: builder/ebssurrogate/step_snapshot_volumes.go (hunk 1 above)
Comment (gnought) [reply builder/ebssurrogate/step_snapshot_volumes.go]: Thanks for the comment.
The fix is pushed.

File: builder/ebsvolume/step_snapshot_ebs_volumes.go (hunk 2 above)
Comment (gnought) [reply builder/ebsvolume/step_snapshot_ebs_volumes.go]: Thanks for the comment.
The fix is pushed.

**Comment by gnought** [comment]
signed CLA and fixes are pushed. Please review. :)

**Comment by lbajolet-hashicorp** [comment]
Hi @gnought,

I see I misunderstood something in my previous review: the description you added was already the default (which was not changeable in the current state), so we should continue forwarding it, apologies for missing that.
The PR in its previous state was good, no need for the rerolls; given they're separate commits, I'll force-push this branch again without them, and merge as soon as tests go green.

Thanks for the update!

**Pull request merged**